google/resource_container_cluster.go

```diff
@@ -1908,6 +1908,24 @@ func resourceContainerClusterStateImporter(d *schema.ResourceData, meta interfac
 	d.Set("name", clusterName)
 	d.SetId(clusterName)
 
+	if config.WaitOnImport {
+		err := resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+			name := containerClusterFullName(project, location, clusterName)
+			cluster, err := config.clientContainerBeta.Projects.Locations.Clusters.Get(name).Do()
+			if err != nil {
+				return resource.NonRetryableError(err)
+			}
+			if cluster.Status != "RUNNING" {
```

Comment:
A comment from Arpit:

>Sometimes we've seen clusters in Error/Failed state where Get() would work but state is never going to change.
It might be better to fail the import right away.